src/log.jl

```diff
@@ -193,10 +193,26 @@ function validate_circuit_decomposition(data::Dict{String, Any}, id_sequence::Ar
         target_gate = QCO.real_to_complex_gate(data["target_gate"])
     end
     
-    if (data["decomposition_type"] in ["exact_optimal", "exact_feasible"]) && (!isapprox(M_sol, target_gate, atol = 1E-4))
-        Memento.error(_LOGGER, "Decomposition is not valid: Problem may be infeasible")
+    global_phase = 1
+
+    if data["decomposition_type"] in ["exact_optimal_global_phase"]
+        n_r = size(target_gate)[1]
```

Comment:
Could these refs be directly obtained using your helper function in utility.jl?

Reply:
changed so this uses: _get_ref_nonzero_index_of_original_target